### PR TITLE
Use lr_scheduler for ImageNet training

### PR DESCRIPTION
**Pull request by pmeier**
This implements the usage of a `torch.optim.lr_scheduler` to take care of the learning rate adjustments. It furthermore deletes the `checkpoint` after it is loaded to reduce the memory requirement if resuming from a previous run.

### CODE REVIEW

**Comment by soumith** [comment]
it was an explicit choice to not use LR schedulers when we wrote the imagenet example, as this particular user space often wants to tweak learning rates by hand.

Having a level of indirection on how to be able to tweak the LR by hand felt like it would miss the point for those sets of users who will go grappling to other avenues like the forums to find out how to do it.

**Comment by pmeier** [comment]
@soumith Out of curiosity: what purpose does the `adjust_learning_rate` function serve, if we expect the user to adjust the learning rate on his own?

**Comment by soumith** [comment]
@pmeier when I say "tweak the LR by hand", I mean "write the code by hand", rather than relying on an opaque class. By having `adjust_learning_rate` close the the rest of the code, in the same file, the user can see and modify immediately the function
